### DISCUSSION TITLE
proposal: more strict useState types

### DISCUSSION THREAD

**Discussion by Shinyaigeek** [nanojsx/nano#23] [Ideas]
## Motivation

In React, [state, setState]`s type returned by useState hooks is inferred with useState's argument or type Parameter.

```typescript
const [value, setValue] = useState(0);
// value is infered as number, and setValue is infered as (value: number) => void

const [value, setValue] = useState<string>(0);
// type error happens
```

but in nano, 
[state, setState]'s value is inferred as any. 

IMO, more strict type will lead better developer experience, so I propose that more strict useState's type.

If this feature is needed, can I work on this?

**Comment by yandeu** [comment]
You're right. Feel free to implement it :)

**Reply by Shinyaigeek** [reply]
Thank you for replying! I did in #24 👍
